### DISCUSSION TITLE
getting error Context: - server returned an error response: error code -32000: insufficient funds for gas * price + value: balance 107279651755905553, tx cost 157805639178998580, overshot 50525987423093027

### DISCUSSION THREAD

**Discussion by arod12345** [Cyfrin/foundry-full-course-cu#2605] [General]
i am getting this insufficient funds for gas error while i clearly have enough funds in my account 
Here is my SimpleStorage.sol file 
```
// I'm a comment!
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

// pragma solidity ^0.8.0;
// pragma solidity >=0.8.0 <0.9.0;

contract SimpleStorage {
    uint256 myFavoriteNumber;
    struct Person {
        uint256 favoriteNumber;
        string name;
    }
    // uint256[] public anArray;
    Person[] public listOfPeople;

    mapping(string => uint256) public nameToFavoriteNumber;

    function store(uint256 _favoriteNumber) public {
        myFavoriteNumber = _favoriteNumber;
    }

    function retrieve() public view returns (uint256) {
        return myFavoriteNumber;
    }

    function addPerson(string memory _name, uint256 _favoriteNumber) public {
        listOfPeople.push(Person(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber;
    }
}
```
my script 

```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {Script} from "forge-std/Script.sol";
import {SimpleStorage} from "../src/SimpleStorge.sol";

contract DeploySimpleStorage is Script {
    function run() external returns (SimpleStorage) {
        vm.startBroadcast();
        SimpleStorage simpleStorage = new SimpleStorage();
        vm.stopBroadcast();
        return simpleStorage;
    }
}
```
and my foundry command 

```
forge script script/DeploySimpleStorage.s.sol --rpc-url $SEPOLIA_RPC_URL  --broadcast   --private-key  $PRIVATE_KEY

```
and my metamask balance is 

![Screenshot 2024-10-03 000659](https://github.com/user-attachments/assets/fbb39f4b-3cb1-4b3d-8b15-da6a3a03d46f)

**Comment by EngrPips** [comment]
Hello @arod12345, Please get more sepolia ETH from chainlink faucet https://faucets.chain.link/sepolia, then retry the deployment process

**Reply by arod12345** [reply]
still couldn't deploy it after getting another 0.1  
Error: 
Failed to send transaction

Context:
- server returned an error response: error code -32000: insufficient funds for gas * price + value: balance 203622218985781553, tx cost 216864422917175880, overshot 13242203931394327

**Reply by arod12345** [reply]
I had the same issue in remix and now its same all thing happening in foundry if anyone could help me it would be a bliss i am kind of giving up i couldn't find a solution

**Reply by EngrPips** [reply]
So we are discussing this already on Discord.

**Reply by arod12345** [reply]
yeah

**Reply by melihxss** [reply]
did you solve this?

**Reply by EngrPips** [reply]
I believe so @arod12345. How did you end up solving this issue? I can't remember correctly

**Reply by rohansai2001** [reply]
Hey @arod12345 @EngrPips has anyone resolved the issue yet ?

**Reply by EngrPips** [reply]
This error basically means you don't have enough funds to cover the transaction fee. what project are you currently woking on and are you working on Remix or locally on `VScode`?

> Hey @arod12345 @EngrPips has anyone resolved the issue yet ?

**Reply by arod12345** [reply]
> Hey @arod12345 @EngrPips has anyone resolved the issue yet ?

yeah it means it you don't have enough funds for gas make sure to add more funds

**Reply by EngrPips** [reply]
Interesting.
